### ISSUE TITLE
Proposal - support swagger-codegn for REST services on f8

### ISSUE THREAD

**Issue by stevef1uk** [fabric8io/ipaas-quickstarts#1065]
As Swagger is the defacto tool to design REST APIs and swagger-codegen provides code generation for client and server side (including Java JAX RS, which works well with maven) it would be nice to be able to easily run the generated code on f8. Then you could use APIMan to expose the APIs apply policies over them etc.

I have looked at the generated code from Swagger-codegen and the sample Ipass-quickstarts REST code and there is a bit of a gap to overcome. Some form of tooling or swagger-codegen would be very nice.

**Comment by KurtStam** [comment]
Related to https://github.com/fabric8io/fabric8/issues/5463

**Comment by stevef1uk** [comment]
I did put up an example project of this working on fabric8 if you want to take a look:

https://github.com/stevef1uk/ipaas-quickstarts.git

in quick start -> java -> camel-example-swagger-java

Looking at the git repo I don't seem to have added the generated JAXRS code. I will take a look at that.

**Comment by KurtStam** [comment]
Thanks @stevef1uk! Your example generates the server side right? In that use case where would the user obtain the swagger document? I can see the case for generating the client side based on a swagger document  obtained from a service, but yours is opposite :).

**Comment by stevef1uk** [comment]
Yes, it is simply a server side RESTful service creased by swagger-code-gen that contains the full swagger documentation running on fabric8.
